aus_council_scrapers/base.py

```diff
@@ -21,132 +21,14 @@
-            return None
-
-    @property
-    def cleaned_date(self) -> datetime.date:
-        if not self.date:
-            raise ValueError("Date is required")
-
-        try:
-            if not self._cleaned_date:
-                self._cleaned_date = parse_date(self.date, fuzzy=True).date()
-        except Exception as e:
-            raise ValueError(f"Could not parse date {self.date}")
-
-        return self._cleaned_date
-
-    @property
-    def cleaned_location(self) -> Optional[str]:
-        if not self.location or self.location.isspace():
-            return None
-
-        cleaned = self.location.replace(r"\w", " ").strip().lower()
-
-        # Remove council chambers string from location
-        council_chamber_regex = re.compile(r"^council\s?chambers?,?", re.IGNORECASE)
-        cleaned = council_chamber_regex.sub("", cleaned)
-
-        if cleaned == "":
-            return None
-
-        return " ".join((word.capitalize() for word in cleaned.split()))
-
-    def check_required_properties(self, state: str) -> None:
-        if not self.name or self.name.isspace():
-            raise ValueError(f"No name found")
-        if not self.download_url or self.download_url.isspace():
-            raise ValueError(f"No download URL found")
-        if not self.webpage_url or self.webpage_url.isspace():
-            raise ValueError(f"No webpage URL found")
-
-        # cleaned date check happens in the property getter
-        _ = self.cleaned_date
-
-        # Check if date is in the past
-        # TODO: Do we want to add this check to make sure we're not scraping meetings that happened in the past?
-        # if self.is_date_in_past(state):
-        #     raise ValueError(f"Meeting date is in the past")
-
-    def add_default_values(self, default_name, default_time, default_location):
-        if not self.name and default_name:
-            self.name = default_name
-        if not self.time and default_time:
-            self.time = default_time
-        if not self.cleaned_location and default_location:
-            self.location = default_location
-
-    def is_date_in_past(self, state: str) -> bool:
-        timezone = pytz.timezone(TIMEZONES_BY_STATE[state.upper()])
-        today = datetime.datetime.now(timezone).date()
-        return self.cleaned_date < today
-
-    def __str__(self):
-        return json.dumps(self.to_dict(), indent=2)
-
-    def to_dict(self):
-        return {
-            "name": self.name,
-            "date": self.date,
-            "time": self.time,
-            "location": self.location,
-            "webpage_url": self.webpage_url,
-            "download_url": self.download_url,
-        }
-
-    @staticmethod
-    def from_dict(d):
-        return ScraperReturn(
-            name=d["name"],
-            date=d["date"],
-            time=d["time"],
-            webpage_url=d["webpage_url"],
-            download_url=d["download_url"],
-            location=d.get("location"),
-        )
```

Comment:
I've realised adding variants to this data type makes this specifically a pain in the ass.

there's `asdict` from `import dataclass` but I am unsure if we're saving this to disc or not and `from_dict` needs to work with existing data saved to disc.